compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -441,6 +441,7 @@ BLangVariableReference addInputFunction(BLangBlockStmt blockStmt, BLangInputClau
                 types.getSafeType(valueAccessExpr.expr.type, true, false));
         VariableDefinitionNode variableDefinitionNode = inputClause.variableDefinitionNode;
         BLangVariable variable = (BLangVariable) variableDefinitionNode.getVariable();
+        setSymbolOwner(variable, env.scope.owner);
```

Comment:
Note: Stmts like `from Employee emp in employees` get's desugared and moved into a `lambda {Employee emp = <Employee> $frame$.value; $frame$.emp = emp; Return: $frame$}`. Therefore, the owner of `emp` should be changed.